packages/backend-common/src/reading/GithubUrlReader.ts

```diff
@@ -229,6 +234,99 @@ export class GithubUrlReader implements UrlReader {
     throw new Error(message);
   }
 
+  private async getRepositoryArchive(
+    repoUrl: string,
+    branchName: string,
+  ): Promise<Response> {
+    return fetch(new URL(`${repoUrl}/archive/${branchName}.tar.gz`).toString());
+  }
+
+  private writeBufferToFile(
+    filePath: string,
+    content: Buffer,
+  ): Promise<Error | undefined> {
+    return new Promise((resolve, reject) => {
+      fs.mkdirSync(path.join(filePath, '../'), { recursive: true });
+      fs.writeFile(filePath, content.toString(), error => {
```

Comment:
There's the fs-extra package with async variants of these functions so you can just use them with await instead of wrapping in a promise